stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -107,29 +107,65 @@ end
 openblas_get_config() = strip(unsafe_string(ccall((@blasfunc(openblas_get_config), libblas), Ptr{UInt8}, () )))
 
 """
-    set_num_threads(n)
+    set_num_threads(n::Integer)
+    set_num_threads(::Nothing)
+
+Set the number of threads the BLAS library should use equal to `n::Integer`.
+If `nothing` is passed to this function, julia tries to figure out the optimial number of threads.
+The exact heuristic is an implementation detail.
 
-Set the number of threads the BLAS library should use.
+On exotic variants of `BLAS` this function can fail, which is indicated by returning `nothing`.
 """
 function set_num_threads(n::Integer)
     blas = vendor()
-    if blas === :openblas
-        return ccall((:openblas_set_num_threads, libblas), Cvoid, (Int32,), n)
-    elseif blas === :openblas64
-        return ccall((:openblas_set_num_threads64_, libblas), Cvoid, (Int32,), n)
+    if blas === :openblas || blas == :openblas64
+        return ccall((@blasfunc(openblas_set_num_threads), libblas), Cvoid, (Int32,), n)
     elseif blas === :mkl
         # MKL may let us set the number of threads in several ways
         return ccall((:MKL_Set_Num_Threads, libblas), Cvoid, (Cint,), n)
-    end
-
-    # OSX BLAS looks at an environment variable
-    @static if Sys.isapple()
+    elseif Sys.isapple()
+        # OSX BLAS looks at an environment variable
         ENV["VECLIB_MAXIMUM_THREADS"] = n
+    else
+        @warn "Failed to set number of BLAS threads." maxlog=1
     end
 
     return nothing
 end
 
+function set_num_threads(::Nothing)
+    n = max(1, Sys.CPU_THREADS ÷ 2)
+    set_num_threads(n)
+end
+
+"""
+    get_num_threads()
+
+Get the number of threads the BLAS library is using.
+
+On exotic variants of `BLAS` this function can fail, which is indicated by returning `nothing`.
+"""
+function get_num_threads()
+    blas = LinearAlgebra.BLAS.vendor()
+    if blas === :openblas || blas === :openblas64
+        return ccall((@blasfunc(openblas_get_num_threads), libblas), Cint, ())
+    elseif blas == :mkl
+        return ccall((:mkl_get_max_threads, libblas), Cint, ())
+    elseif Sys.isapple()
+        key = "VECLIB_MAXIMUM_THREADS"
+        s = get(ENV, key, "")
+        nt = Base.tryparse(Cint, s)
+        if nt === nothing
+            @warn "Failed to read environment variable $key" maxlog=1
+        else
+            return nt
+        end
+    end
+    ret = nothing
+    @warn "Could not get number of BLAS threads. Returning `$ret` instead." maxlog=1
```

Comment:
```suggestion
    @warn "Could not get number of BLAS threads. Returning `nothing` instead." maxlog=1
```

Reply:
since `ret` is always nothing in this case we can drop the interpolation.

Reply:
If we go with it, we should just do `return nothing`. See: https://github.com/JuliaLang/julia/pull/36360#discussion_r443108316